sdk-core/src/main/java/care/data4life/sdk/tag/TagEncryptionService.kt

```diff
@@ -40,6 +43,19 @@ class TagEncryptionService @JvmOverloads constructor(
                 .blockingGet()
     }
 
+    @Throws(D4LException::class)
+    private fun encryptTag(key: GCKey, tag: String): Single<String> {
```

Comment:
Even if you just moved the code, a shrinked version avoiding Single as suggestion. Error handling needs to be pushed up the stream or fail in place:

```
    @Throws(D4LException::class)
    private fun encryptTag(key: GCKey, tag: String): String {
        return listOf(tag.toByteArray())
                .map { plain -> cryptoService.symEncrypt(key, plain, IV) }
                .map { data -> base64.encodeToString(data) }
                .first()
    }
```

Reply:
Very good point...and I guess I actually what was the original intent now. The reverse function can be altered in the same way. Therefore we can totally get rid of RxJava in this class.
However the crypto service methods should be (in the future) then some sort async function.